src/DeploymentTemplate.ts

```diff
@@ -360,7 +313,7 @@ export class DeploymentTemplate {
         const apiProfileString = `(profile=${this.apiProfile || 'none'})`.toLowerCase();
 
         // Collect all resources used
-        const resources: Json.ArrayValue | undefined = this._topLevelValue ? Json.asArrayValue(this._topLevelValue.getPropertyValue(templateKeys.resources)) : undefined;
+        const resources: Json.ArrayValue | undefined = this.topLevelValue ? Json.asArrayValue(this.topLevelValue.getPropertyValue(templateKeys.resources)) : undefined;
```

Comment:
>Json.ArrayValue | undefined = this.topLevelValue ? Json.asArrayValue(this.topLevelValue.getPropertyValue(templateKeys.resources)) : undefined; [](start = 25, length = 142)

We have a helper for this now, right?

Reply:
Yes, thank heavens, now goes to simply:

return this.topLevelValue?.getPropertyValue(templateKeys.resources)?.asArrayValue;

Reply:
Might start taking a pass later at simplifying all this.